packages/dds/tree/src/dependency-tracking/dependencies.ts

```diff
@@ -22,6 +22,20 @@ export interface NamedComputation {
 	 * Use when measuring / debugging / logging computation costs, invalidation etc.
 	 */
 	readonly computationName: string;
+
+    /**
+     * Lists the currently subscribed set of Dependees.
+     * This is exposed to allow tooling to inspect the dependency graph,
+     * and should not be needed for regular functionality.
+     */
+    listDependees?(): Iterable<Dependee>;
+
+    /**
+     * Lists the currently subscribed set of Dependent.
+     * This is exposed to allow tooling to inspect the dependency graph,
+     * and should not be needed for regular functionality.
+     */
+    listDependents?(): Iterable<Dependent>;
```

Comment:
A couple minor suggestions:
- Any reason these aren't properties?
- Perhaps use symbols to make them less discoverable?

Also, in the FRP system I worked on previously, we switch from dependee/dependent to producer/consumer, as for some reason this was easier for newcomers.

Reply:
A couple reasons not to use properties here:
- optional properties tend to imply that there is no value when undefined, rather than the functionality to expose the value is not available (which is whats desired here).
- We want to explicitly imply that this might not be free to compute: it should be implemented if its practical to do so, not just if its efficient since its not used in performance sensitive ways.
- Implementations of these types frequently already have a private field named dependees or dependents, and typescript does not provider a way to have a private field that is exported with a less specific type (ex: read only or Iterable instead of Set).

Making them optional means people can't accidently depend on them without getting an error: I think thats enough.
If I wasn't putting them on the base type here, then symbols would be a great option to make the effective down casting safter. As is I don't think symbols offers much value, and makes them harder to call in the debugger.